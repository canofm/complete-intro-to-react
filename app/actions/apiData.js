import axios from 'axios';

export const ADD_API_DATA = 'ADD_API_DATA';

export const addAPIData = apiData => ({ type: ADD_API_DATA, payload: apiData });

export const getAPIDetails = imdbID => dispatch =>
  axios
    .get(`http://localhost:3000/${imdbID}`)
    .then(response => dispatch(addAPIData(response.data)))
    .catch(error => console.log('axios error: ', error)); // eslint-disable-line no-console
