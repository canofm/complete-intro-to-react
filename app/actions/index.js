const SET_SEARCH_TERM = 'SET_SEARCH_TERM';

const setSearchTerm = searchTerm => ({ type: SET_SEARCH_TERM, payload: searchTerm });

export { SET_SEARCH_TERM };
export { setSearchTerm };
