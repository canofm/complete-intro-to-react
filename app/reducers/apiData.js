import { ADD_API_DATA } from '../actions';

const apiData = (state = {}, action) => {
  if (action.type === ADD_API_DATA) {
    return Object.assign({}, state, { [action.payload.imdbID]: action.payload });
  }
  return state;
};

export default apiData;
