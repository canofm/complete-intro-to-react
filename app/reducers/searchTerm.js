import { SET_SEARCH_TERM } from '../actions';

const searchTerm = (state = '', action) => {
  if (action.type === SET_SEARCH_TERM) {
    return action.payload;
  }
  return state;
};

export default searchTerm;
