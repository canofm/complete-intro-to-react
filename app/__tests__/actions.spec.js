import { setSearchTerm } from '../actions/searchTerm';
import { addAPIData } from '../actions/apiData';

test('setSearchTerm', () => {
  expect(setSearchTerm('Black')).toMatchSnapshot();
});

test('addAPIData', () => {
  const blackMirror = {
    title: 'Black Mirror',
    year: '2011â€“',
    description: 'A television anthology series that shows the dark side of life and technology.',
    poster: 'bm.jpg',
    imdbID: 'tt2085059',
    trailer: 'jDiYGjp5iFg',
    rating: '9.7'
  };

  expect(addAPIData(blackMirror)).toMatchSnapshot();
});
