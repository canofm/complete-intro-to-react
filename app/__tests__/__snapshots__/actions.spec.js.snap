// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "A television anthology series that shows the dark side of life and technology.",
    "imdbID": "tt2085059",
    "poster": "bm.jpg",
    "rating": "9.7",
    "title": "Black Mirror",
    "trailer": "jDiYGjp5iFg",
    "year": "2011â€“",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "Black",
  "type": "SET_SEARCH_TERM",
}
`;
